src/Adapter/Product/CommandHandler/UpdateProductBasicInformationHandler.php

```diff
@@ -110,7 +110,7 @@ private function fillUpdatableProperties(
             // Go through all the product languages and make sure name is filled for each of them
             $productLanguages = array_keys($product->name);
             foreach ($productLanguages as $languageId) {
-                if (empty($product->name[$languageId]) && empty($localizedNames[$languageId])) {
```

Comment:
when product name already existed before - this statement failed therefor not setting the default name.